Mathlib/Tactic/Set.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2022 Ian Benway. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Ian Benway.
+-/
+import Lean
+namespace Mathlib.Tactic
+open Lean Elab.Tactic Meta
+
+private def defineV (nm : Name) (tp : Expr) (tme : Expr) : TacticM Unit := do
+  liftMetaTactic1 fun mvarId => do
+    let h2 ← (define mvarId nm tp tme)
+    let (h, h2) ← intro1P h2
+    withMVarContext h2 do
+      return h2
```

Comment:
I have an open PR with a function that's very similar to this one. Once both PRs are merged I can do a mini refactor so that both implementations use an unified function